tests/python_client/testcases/test_search.py

```diff
@@ -3001,42 +3011,53 @@ def test_search_output_field_vector_with_different_dim(self, dim):
         for i in range(default_limit):
             assert len(res[0][i].entity.float_vector) == len(data[field_name][res[0][i].id])
 
-    @pytest.mark.tags(CaseLabel.L1)
-    @pytest.mark.parametrize("dim", [32, 128, 768])
-    def test_search_output_field_vector_with_different_dim(self, dim):
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("metrics", ct.float_metrics)
+    def test_search_output_field_vector_different_metric_type(self, metrics):
         """
-        target: test search with output vector field after binary index
+        target: test search with output vector field
         method: 1. create a collection and insert data
                 2. create index and load
                 3. search with output field vector
                 4. check the result vectors should be equal to the inserted
         expected: search success
         """
         # 1. create a collection and insert data
-        collection_w = self.init_collection_general(prefix, is_index=False, dim=dim)[0]
-        data = cf.gen_default_dataframe_data(dim=dim)
+        collection_w = self.init_collection_general(prefix, is_index=False)[0]
+        data = cf.gen_default_dataframe_data()
         collection_w.insert(data)
 
         # 2. create index and load
-        index_params = {"index_type": "IVF_FLAT", "params": {"nlist": 128}, "metric_type": "L2"}
+        index_params = {"index_type": "FLAT", "params": {"nlist": 128}, "metric_type": metrics}
         collection_w.create_index("float_vector", index_params)
         collection_w.load()
 
         # 3. search with output field vector
-        vectors = cf.gen_vectors(default_nq, dim=dim)
-        res = collection_w.search(vectors[:default_nq], default_search_field,
+        search_vectors = cf.gen_vectors(default_nq, dim=default_dim)
+        res = collection_w.search(search_vectors[:default_nq], default_search_field,
                                   default_search_params, default_limit, default_search_exp,
                                   output_fields=[field_name],
                                   check_task=CheckTasks.check_search_results,
                                   check_items={"nq": default_nq,
                                                "limit": default_limit})[0]
 
         # 4. check the result vectors should be equal to the inserted
-        log.info(res[0][0].id)
-        log.info(res[0][0].entity.float_vector)
-        log.info(data['float_vector'][0])
         for i in range(default_limit):
             assert len(res[0][i].entity.float_vector) == len(data[field_name][res[0][i].id])
+        getcontext().rounding = getattr(decimal, 'ROUND_HALF_UP')
+        for _id in range(default_limit):
+            for i in range(default_dim):
```

Comment:
why we need to assert for every dim in range(default_dim)

Reply:
> why we need to assert for every dim in range(default_dim)

I'll try to find an easier way.